### PR TITLE
Add dbWatch template hooks to server models

### PR DESCRIPTION
**Pull request by Setogit**
Template dbWatcher hooks added to server models for initial end-to-end testing.  Existing loopback connector based notification is still intact.

### CODE REVIEW

File: server/models/agent-trace.js [hunk 1]
```diff
@@ -2,7 +2,14 @@ var async = require('async');
 var debug = require('debug')('strong-mesh-models:server:agent-trace');
 var deleteOld = require('./delete-old');
 
-module.exports = function extendAgentTrace(AgentTrace) {
+exports = module.exports = extendAgentTrace;
+exports.modelWatcher = modelWatcher;
+
+function modelWatcher(msg) {
+  console.log('------------------------ agenttrace:', msg);
```
Comment (kraman) [inline server/models/agent-trace.js]: Left over debug statement?

File: server/boot/extend-models.js [hunk 2]
```diff
@@ -18,4 +20,27 @@ module.exports = function(server) {
...
+    {name: 'executor', instance: executor},
+    {name: 'expressusagerecord', instance: expressUsageRecord},
+    {name: 'profiledata', instance: profileData},
+    {name: 'serviceinstance', instance: serviceInstance},
+    {name: 'serverservice', instance: serverService},
+    {name: 'servicenetric', instance: serviceMetric},
+    {name: 'serviceprocess', instance: serviceProcess},
+    {name: 'gateway', instance: gateway},
+  ];
+  server.on('ready', function(dbWatcher) {
```
Comment (kraman) [inline server/boot/extend-models.js]: Where does this event come from and at what point is it emitted? The watchers will need to be set up after DB, Driver etc are crated but before the driver and gateway re-connect calls. Perhaps this should be an method on the mesh server that can be called when we need it?

Comment (kraman) [inline server/boot/extend-models.js]: This should probably also check if dbWatcher is passed in. It would be null for non-postgres cases...right?

**Comment by Setogit** [comment]
Moved the dbWatch initialization to ServiceManager.  serviceDestroy change is still useful.

**Pull request closed without merging**